src/internal/jacobian.jl

```diff
@@ -63,31 +62,13 @@ function JacobianCache(prob, alg, f::F, fu_, u, p; stats, autodiff = nothing,
 
     if !has_analytic_jac && needs_jac
         autodiff = construct_concrete_adtype(f, autodiff)
-        using_sparsedifftools = autodiff isa StructuredMatrixAutodiff
-        # SparseMatrixColorings can't handle structured matrices
-        if using_sparsedifftools
-            di_extras = nothing
-            uf = JacobianWrapper{iip}(f, p)
-            sdifft_extras = if iip
-                sparse_jacobian_cache(
-                    autodiff.autodiff, autodiff.sparsity_detection, uf, fu, u)
-            else
-                sparse_jacobian_cache(autodiff.autodiff, autodiff.sparsity_detection,
-                    uf, __maybe_mutable(u, autodiff); fx = fu)
-            end
-            autodiff = autodiff.autodiff # For saving we unwrap
+        di_extras = if iip
+            DI.prepare_jacobian(f, fu, autodiff, u, Constant(prob.p))
         else
-            sdifft_extras = nothing
-            di_extras = if iip
-                DI.prepare_jacobian(f, fu, autodiff, u, Constant(prob.p))
-            else
-                DI.prepare_jacobian(f, autodiff, u, Constant(prob.p))
-            end
+            DI.prepare_jacobian(f, autodiff, u, Constant(prob.p))
         end
     else
-        using_sparsedifftools = false
         di_extras = nothing
```

Comment:
In which cases do we not prepare the jacobian? With DI, if you forget to do that, you'll recompute the sparsity pattern, the coloring and the basis vectors every single time

Reply:
accidental, removed

Reply:
you don't prepare the jacobian if the solver doesn't need the jacobian and the user didn't provide a jacobian function. In these cases, the nonlinear solver is doing everything with vjps and jvps anyway so you don't need this info.